src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,150 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.IO;
+using System.Collections.Specialized;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices
+{
+    public class PeekDefinition
+    {
+        private ConnectionInfo connectionInfo;
+        private string tempPath;
+        public PeekDefinition(ConnectionInfo connInfo)
+        {
+            connectionInfo = connInfo;
+            tempPath = Path.GetTempPath();
+        }
+
+        /// <summary>
+        /// Script a table using SMO and write to a file.
+        /// </summary>
+        /// <param name="tableName">Table name</param>
+        /// <returns>Location object representing URI and range of the script file</returns>
+        internal Location[] GetTableDefinition(string tableName)
+        {
+            if (this.connectionInfo.SqlConnection != null)
+            {
+                Server server = new Server(this.connectionInfo.SqlConnection.DataSource);
+                Database database = server.Databases[this.connectionInfo.SqlConnection.Database];
+                Table table = database.Tables[tableName];
+                string tempFileName = tempPath + tableName + ".sql";
+
+                if (table != null)
+                {
+                    using (StreamWriter scriptFile = new StreamWriter(File.Open(tempFileName, FileMode.Create, FileAccess.ReadWrite)))
+                    {
+                        StringCollection scripts = table.Script();
+                        foreach (string script in scripts)
+                        {
+                            if (script.Contains("CREATE TABLE"))
+                            {
+                                scriptFile.WriteLine(script);
```

Comment:
There can be only one CREATE I think. I'm looking into scripting options that would let us remove this check.